pkg/elemental/components/ImportMultiple.vue

```diff
@@ -0,0 +1,218 @@
+<script>
+import { mapGetters } from 'vuex';
+import { Card } from '@components/Card';
+import { Banner } from '@components/Banner';
+import Loading from '@shell/components/Loading';
+import FileSelector from '@shell/components/form/FileSelector';
+import AsyncButton from '@shell/components/AsyncButton';
+import LabeledSelect from '@shell/components/form/LabeledSelect';
+import SortableTable from '@shell/components/SortableTable';
+import { sortBy } from '@shell/utils/sort';
+import { exceptionToErrorsArray } from '@shell/utils/error';
+import { NAMESPACE } from '@shell/config/types';
+import {
+  NAME as NAME_COL, STATE, TYPE, NAMESPACE as NAMESPACE_COL, AGE
+} from '@shell/config/table-headers';
+
+export default {
+  components: {
+    AsyncButton,
+    Banner,
+    Card,
+    Loading,
+    FileSelector,
+    LabeledSelect,
+    SortableTable
+  },
+
+  props: {
+    defaultNamespace: {
+      type:    String,
+      default: 'default'
+    },
+  },
+
+  async fetch() {
+    this.allNamespaces = await this.$store.dispatch('cluster/findAll', { type: NAMESPACE, opt: { url: 'namespaces' } });
+  },
+
+  data() {
+    return {
+      allYamls:      [],
+      allNamespaces: null,
+      errors:        null,
+      rows:          null,
+      done:          false,
+    };
+  },
+
+  computed: {
+    ...mapGetters(['currentCluster']),
+
+    namespaceOptions() {
+      const out = this.allNamespaces.map((obj) => {
+        return {
+          label: obj.name,
+          value: obj.name,
+        };
+      });
+
+      return sortBy(out, 'label');
+    },
+
+    headers() {
+      return [
+        STATE,
+        TYPE,
+        NAME_COL,
+        NAMESPACE_COL,
+        AGE
+      ];
+    },
+  },
+
+  methods: {
+    close() {
+      this.$emit('close');
+    },
+
+    onFileSelected(filesContent) {
+      this.allYamls = [];
+      this.allYamls = filesContent;
+    },
+
+    handleMultipleImports(btnCb) {
+      const promises = [];
+
+      this.allYamls.forEach(file => promises.push(this.importYaml(file.value)));
+
+      Promise.allSettled(promises).then((response) => {
+        const rows = [];
+        const errors = [];
+
+        response.forEach((res) => {
+          if (res.value && res.value[0]) {
+            rows.push(res.value[0]);
+          } else {
+            errors.push(res.value.err);
+          }
+        });
+
+        if (errors.length) {
+          const errorMsg = `${ errors.length } out of ${ response.length } YAML files could not be applied successfully.`;
+
+          this.errors = exceptionToErrorsArray(errorMsg);
```

Comment:
Normally when there are errors resulted from an AsyncButton action the btn callback should be passed `false` (see usages of `btnCb(false)`)